src/language-server/config/loadConfig.ts

```diff
@@ -206,11 +206,13 @@ export async function loadConfig({
   // selectively apply defaults when loading the config
   // this is just the includes/excludes defaults.
   // These need to go on _all_ configs. That's why this is last.
-  if (config.client) config = merge({ client: DefaultClientConfig }, config);
```

Comment:
Lodash `merge` deep-merges into the first argument, so this would actually modify `DefaultClientConfig`, which would then provide defaults to the next config file that would be loaded.